### PR TITLE
vlinsert: parse loki JSON body if msg_fields are defined

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Parse JSON from Loki message field if at least one msg_field is defined

related issues:
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7932

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Comment by jiekun** [comment]
Hi Will this PR solve https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7932 ?

**Comment by AndrewChubatiuk** [comment]
@jiekun, yes
but now user can apply [`nest`](https://docs.fluentbit.io/manual/pipeline/filters/nest) filter to move nested values from `log` field to a top level with `log_` prefix and then use [`modify`](https://docs.fluentbit.io/manual/pipeline/filters/modify) filter to rename `log_message` to `log`

File: app/vlinsert/loki/loki_json.go [hunk 1]
```diff
@@ -153,16 +153,30 @@ func parseJSONRequest(data []byte, lmp insertutils.LogMessageProcessor, useDefau
 			}
 
 			// parse log message
-			msg, err := lineA[1].StringBytes()
-			if err != nil {
-				return fmt.Errorf("unexpected log message type for %q; want string", lineA[1])
+			msgV := lineA[1]
+			msgT := msgV.Type()
+			switch {
+			case msgT == fastjson.TypeString || len(msgFields) == 0:
```
Comment (f41gh7) [inline app/vlinsert/loki/loki_json.go:159]: According to the Loki protocol format - `log line` must be a string. 

 https://github.com/grafana/loki/blob/ada4b7b8713385fbe9f5984a5a0aaaddf1a7b851/pkg/push/types.go#L26

Comment (AndrewChubatiuk) [reply app/vlinsert/loki/loki_json.go:159]: yes, indeed, closing PR

**Pull request closed without merging**